### DISCUSSION TITLE
Autogenerating parameters in YAML for github action dispatch events

### DISCUSSION THREAD

**Discussion by MaddoScientisto** [nuke-build/nuke#1005] [Q&A] [answered]
The github actions API allows triggering builds through the API like this:

```bash
curl \
  -X POST \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/actions/workflows/WORKFLOW_ID/dispatches \
  -d '{"ref":"topic-branch","inputs":{"name":"Mona the Octocat","home":"San Francisco, CA"}}'
```

How can these inputs be generated through the yaml generator?
I tried this
```C#
[GitHubActions("continuous",
               GitHubActionsImage.WindowsServer2022,
               InvokedTargets = new[] { nameof(BuildSetup) },
               On = new[] { GitHubActionsTrigger.WorkflowDispatch },
               EnableGitHubToken = true, OnWorkflowDispatchOptionalInputs = new []{"name", "home"},
)]
```

in the hope that would send that name parameter to the build script but all I get is a very unhelpful error when trying a build and the workflow yml is generated.
```
One or more errors occurred. (Process 'dotnet.exe' exited with code -1.
   > "C:\Program Files\dotnet\dotnet.exe" C:\Users\Maddo\Documents\GitHub\Plugin\build\bin\Debug\_build.dll --generate-configuration GitHubActions_continuous --host GitHubActions
   @ C:\Users\Maddo\Documents\GitHub\Plugin
)
```

How do I generate these parameters properly?
This is not in the documentation on the site for some reason

**Comment by matkoch** [comment, marked as answer]
Try executing the command that is shown. I haven't looked into that area for a long time.

**Reply by MaddoScientisto** [reply]
In the end I found out that I had to remove the "On" line if I want the OnWorkflowDispatchOptionalInputs to work

**Reply by matkoch** [reply]
@MaddoScientisto would be nice if you could give the project a star 👍
